mcux/middleware/wifi_nxp/sdio_nxp_abs/mlan_sdio.c

```diff
@@ -0,0 +1,161 @@
+/** @file mlan_sdio.c
+ *
+ *  @brief This file provides mlan driver for SDIO
+ *
+ *  Copyright 2008-2024 NXP
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+#include <wmerrno.h>
+#include <fsl_os_abstraction.h>
+#include <board.h>
+#include <wifi_bt_config.h>
+
+#include <mlan_sdio_api.h>
+#include <osa.h>
+#include <fsl_common.h>
+#include <fsl_gpio.h>
```

Comment:
No need for the `fsl_xx` includes here, right?

Reply:
these are needed for sdio case

Reply:
Am I missing somewhere in this file where we use FSL APIs? We *should not* do this- we want to make sure the SDIO code can run on non-nxp platforms, so we should use Zephyr APIS, not SDK APIs.

Reply:
@MaochenWang1 ping on this question

Reply:
Sdio is not used by RW612, but other nxp wifi chip use it.
I check with Sandip about this, and he confirm they use the fsl_xx for sdio case.

Reply:
We cannot use the `fsl_xxx` APIs for the SDIO case. The issue here is that the SDIO stack in Zephyr is generic- IE it can run on non NXP platforms. If we use the `fsl_xxx` APIs for SDIO support, we limit the SDIO WiFi devices to running on NXP platforms that support the `fsl_xxx` APIs. We need to use Zephyr APIs only in our SDIO support files, and not have any specific dependencies on the `fsl_xxx` API layer. This way customers can use the SDIO devices with non-NXP SOCs

Reply:
you mean customer will use our wifi_nxp code and run on non-NXP socs?

Reply:
It's very weird, our wifi_nxp driver is bind with NXP wifi FW(there are many private cmds and events exchange), so I can't image how other non-NXP chips use our wifi_nxp code to build image.
Never mind, I will remove these fsl_xx.

Reply:
removed, please review

Reply:
> It's very weird, our wifi_nxp driver is bind with NXP wifi FW(there are many private cmds and events exchange), so I can't image how other non-NXP chips use our wifi_nxp code to build image.
Never mind, I will remove these fsl_xx.

So what I mean here is that our WiFi driver could be used with a non NXP *host* chip. For example, an STM32 SOC with an SD host controller could be used with the IW416. We want to make sure the WiFi driver *host* code will run on the STM32 device correctly.

Reply:
Even with the binding to the NXP binary blobs, those customers would still be able to 'blob fetch' the blobs.